packages/babel-parser/src/parser/expression.js

```diff
@@ -1214,6 +1214,25 @@ export default class ExpressionParser extends LValParser {
         return node;
       }
 
+      case tt.moduloAssign:
+        if (
+          this.getPluginOption("pipelineOperator", "proposal") === "hack" &&
+          this.getPluginOption("pipelineOperator", "topicToken") === "%"
+        ) {
+          // If we find %= in an expression position, and % could be a topic token,
+          // reparse it as %.
+          // The next readToken() call will start parsing from =.
+
+          this.state.value = "%";
+          this.state.type = tt.modulo;
+          this.state.pos--;
+          this.state.end--;
+          this.state.endLoc.column--;
```

Comment:
should probably check `!isLookahead` before accessing `endLoc`

Reply:
`isLookahead` can never happen here, because lookahead only gets a single token and doesn't advance parsing.

Reply:
Oh I see, lookahead reverts the state back immediately after getting the token, before doing anything with it.